mountpoint-s3/src/inode.rs

```diff
@@ -2797,4 +2809,53 @@ mod tests {
         assert_eq!(file_inodestat.ctime, ts);
         assert_eq!(file_inodestat.mtime, ts);
     }
+
+    #[cfg(feature = "shuttle")]
+    mod shuttle_tests {
+        use mountpoint_s3_client::mock_client::{MockClient, MockClientConfig};
+        use shuttle::{check_pct, check_random, thread};
+        use shuttle::{future::block_on, sync::Arc};
+
+        use super::*;
+
+        #[test]
+        fn test_create_and_forget_race_condition() {
+            async fn test_helper() {
+                let client_config = MockClientConfig {
+                    bucket: "test_bucket".to_string(),
+                    part_size: 1024 * 1024,
+                    ..Default::default()
+                };
+                let client = Arc::new(MockClient::new(client_config));
+
+                let name = "foo";
+                client.add_object(name, b"foo".into());
+
+                let superblock = Arc::new(Superblock::new("test_bucket", &Default::default(), Default::default()));
+
+                let lookup = superblock.lookup(&client, ROOT_INODE_NO, name.as_ref()).await.unwrap();
+                let lookup_count = lookup.inode.inner.sync.read().unwrap().lookup_count;
+                assert_eq!(lookup_count, 1);
+                let ino = lookup.inode.ino();
+
+                let superblock_clone = superblock.clone();
+                let forget_task = thread::spawn(move || {
+                    superblock_clone.forget(ino, 1);
+                });
+
+                let file_name = "bar";
+                superblock
+                    .create(&client, ROOT_INODE_NO, file_name.as_ref(), InodeKind::File)
+                    .await
+                    .unwrap();
+
+                forget_task.join().unwrap();
+                let lookup_count = lookup.inode.inner.sync.read().unwrap().lookup_count;
+                assert_eq!(lookup_count, 0);
+            }
+
+            check_random(|| block_on(test_helper()), 1000);
+            check_pct(|| block_on(test_helper()), 1000, 3);
```

Comment:
I don't know enough about PCT to know if its worth doing both, but I see we do this elsewhere so LGTM :)

Nice that this test case catches the deadlock and now passes.